vars/snykExecute.groovy

```diff
@@ -0,0 +1,101 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.Utils
+import com.sap.piper.mta.MtaMultiplexer
+
+import groovy.transform.Field
+
+@Field def STEP_NAME = 'snykExecute'
+
+@Field Set GENERAL_CONFIG_KEYS = ['snykCredentialsId']
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+    'buildDescriptorFile',
+    'dockerImage',
+    'dockerWorkspace',
+    'exclude',
+    'monitor',
+    'scanType',
+    'snykOrg',
+    'toJson',
+    'toHtml'
+])
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+def call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+        def utils = parameters.juStabUtils ?: new Utils()
+        def script = parameters.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            // check mandatory paramerers
+            .withMandatoryProperty('dockerImage')
+            .withMandatoryProperty('dockerWorkspace')
+            .withMandatoryProperty('snykCredentialsId')
+            .use()
+
+        utils.unstashAll(config.stashContent)
+
+        switch(config.scanType) {
+            case 'mta':
+                def scanJobs = [failFast: false]
+                // create job for each package.json with scanType: 'npm'
+                scanJobs.putAll(MtaMultiplexer.createJobs(
+                    this, parameters, config.exclude, 'Snyk', 'package.json', 'npm'
+                ){options -> snykExecute(options)})
+                // execute scan jobs in parallel
+                parallel scanJobs
+                break
+            case 'npm':
+                // set default file for scanType
+                def path = config.buildDescriptorFile.replace('package.json', '')
+                try{
+                    withCredentials([string(
+                        credentialsId: config.snykCredentialsId,
+                        variable: 'token'
+                    )]) {
+                        dockerExecute(
+                            dockerImage: config.dockerImage,
```

Comment:
Preconfigure node docker image and docker workspace

Reply:
done